EIPS/eip-5753.md

```diff
@@ -0,0 +1,230 @@
+category: ERC
+created: 2022-10-05
+requires: 165, 721
+---
+
+## Abstract
+This standard is an extension of [EIP-721](./eip-721.md). It introduces lockable NFTs. The locked asset can be used in any way except selling and/or transferring it. Owner or operator can lock the token. When a token is being locked, the unlocker address (an EOA or a contract) is set. Only unlocker is able to `unlock` the token. 
+
+## Motivation
+With NFTs, digital objects become digital goods. Verifiably ownable, easily tradable, immutably stored on the blockchain. However, the usability of NFT presently is quite limited. Existing use cases often have poor UX as they are inherited from fungible tokens world.
+
+In DeFi you mostly deal with fungible tokens. There is a UX pattern when you lock your tokens on a service smart contract. For example, if you want to borrow some $DAI, you have to provide some $ETH as collateral for a loan. During the loan period $ETH is being locked into the lending service contract. And it's ok for $ETH and other fungible tokens.
+
+It's different for NFTs. NFTs have plenty of use cases, that require for the NFT to stay on the holder's wallet even when it is used as collateral for a loan. You may want to keep using your NFT as a verified PFP on Twitter. You may want to use it to authorize on Discord server through Collab.land. You may want to use your NFT in a P2E game. And you should be able to do all of this even during the lending period like you are able to live in your house even it is mortgaged.
+
+The initial idea was to just make NFTs that will feature better UX used as collateral. Then it became obvious, that one single locking feature allows for plenty of other use cases, such as:
+* Lending/borrowing NFT without a need for collateral
+* Paying for NFT by installments
+* Safe and convenient usage with hot wallets
+* Non-custodial staking and much more. 
+Every use case can (and some of them are already) be implemented one at a time. My aim however was to come up with a standardized implementation.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+`ERC-721` compliant contracts MAY implement this ERC to provide standard methods of locking and unlocking the token at it's current owner address. 
+If the token is locked, the `getLocked` function MUST return an address, that is able to unlock the token.
+For the tokens that are not locked the `getLocked` function MUST return `address(0)`.
+The user MAY permanently lock the token by stating a smart-contract that does not implement interface to `unlock` function as unlocker.
+
+When the token is locked, all the [EIP-721](./eip-721.md) transfer functions MUST revert, except the msg.sender for the transfer tx is stated as unlocker for this token.
+When the token is locked, [EIP-721](./eip-721.md) `approve` method MUST revert.
```

Comment:
```suggestion
When the token is locked, the [EIP-721](./eip-721.md) `approve` method MUST revert.
```
Should all approvals be cleared while the token is locked?

Reply:
I don't think so. If it is locked without transfer (used as a collateral) we can decide to keep approvals, so when the loan is repaid, there's no need to approve again

Reply:
I meant that the approvals are set to zero while it's locked, and are reset back to the pre-locked value once it's unlocked.